src/openforms/prefill/__init__.py

```diff
@@ -174,15 +173,26 @@ def invoke_plugin(item: Tuple[str, List[str]]) -> Tuple[str, Dict[str, Any]]:
 
 def _extract_prefill_fields(configuration: JSONObject) -> List[Dict[str, str]]:
     prefills = []
-    components = configuration.get("components", [])
 
-    for component in components:
-        if prefill := component.get("prefill"):
-            if prefill.get("plugin"):
-                prefills.append(prefill)
-        else:
-            nested = _extract_prefill_fields(component)
-            prefills += nested
+    if "type" in configuration:
+        fieldType = configuration.get("type")
+
+        if fieldType == "columns":
+            columns = configuration.get("columns", [])
+
+            for column in columns:
+                nested = _extract_prefill_fields(column)
+                prefills += nested
+    else:
+        components = configuration.get("components", [])
+
+        for component in components:
+            if prefill := component.get("prefill"):
+                if prefill.get("plugin"):
+                    prefills.append(prefill)
+            else:
+                nested = _extract_prefill_fields(component)
+                prefills += nested
 
```

Comment:
* Every FormIO component should always have the `type` property - if it's not present and causes tests to fail, then the test data setup is wrong.
* There's some duplicated handling of the FormIO (nested datastructures) here, for which we have  a utility function: `openforms.formio.utils.iter_components`. You can make that recursively loop over the entire configuration/component tree. It already has logic for `columns` type components. This code should have been using that in the first place
* Typically we also add a regression test for bugfixes to show the problem and prove that it was fixed by the patch
* Code style (nitpicking): Python uses `snake_case` rather than `camelCase`

So, collecting all those remarks, I think a better approach is to implement this function as:

```py
from openforms.formio.utils import iter_components

prefills = []
for component in iter_components(configuration, recursive=True):
    if not (prefill := component.get("prefill")):
        continue
    if prefill.get("plugin"):
        prefills.append(prefill)
return prefills
```

And then add a test case in https://github.com/open-formulieren/open-forms/blob/master/src/openforms/prefill/tests/test_prefill_hook.py#L109 for the `columns` type. You can see some example configurations here: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/formio/rendering/tests/test_vanilla_formio_components.py#L110

Reply:
Thanks for the comments. I will update the PR. 

:)